test/with_dummyserver/test_poolmanager.py

```diff
@@ -123,6 +123,17 @@ def test_redirect_cross_host_remove_headers(self):
 
         self.assertNotIn('Authorization', data)
 
+        r = http.request('GET', '%s/redirect' % self.base_url,
+                         fields={'target': '%s/headers' % self.base_url_alt},
+                         headers={'authorization': 'foo'})
+
+        self.assertEqual(r.status, 200)
+
+        data = json.loads(r.data.decode('utf-8'))
+
+        self.assertNotIn('authorization', data)
+        self.assertNotIn('Authorization', data)
```

Comment:
Let's also add a new test for an additional header (eg `X-API-Secret`) and seeing that that header is removed.

Reply:
`test_redirect_cross_host_set_removed_headers`